src/System.Diagnostics.Process/src/System/Diagnostics/Process.FreeBSD.cs

```diff
@@ -64,8 +65,29 @@ public TimeSpan PrivilegedProcessorTime
         }
 
         /// <summary>Gets parent process ID</summary>
-        private int ParentProcessId =>
-            throw new PlatformNotSupportedException();
+        private unsafe int ParentProcessId
+        {
+            get
+            {
+                EnsureState(State.HaveNonExitedId);
+
+                Interop.Process.kinfo_proc* processInfo = null;
+                try
+                {
+                    processInfo = Interop.Process.GetProcInfo(_processId, false, out int count);
+
+                    if (processInfo == null || count <= 0)
```

Comment:
Hmm. I just looked at the implementation of GetProcCount. This code won't be hit if the result would null, so the null check turns out to not be necessary. And, yes, the call can be moved out of the try block, here and elsewhere. There's no benefit to the finally running if it throws.

Reply:
Removed null checks and hoisted out `GetProcInfo` call from try block, in all three places.